mount/mount_linux.go

```diff
@@ -28,14 +29,27 @@ import (
 	"golang.org/x/sys/unix"
 )
 
-var pagesize = 4096
+var (
+	pagesize              = 4096
+	allowedHelperBinaries = []string{"mount.fuse", "mount.fuse3"}
+)
 
 func init() {
 	pagesize = os.Getpagesize()
 }
 
-// Mount to the provided target path
+// Mount to the provided target path.
+//
+// If m.Type starts with "fuse." or "fuse3.", "mount.fuse" or "mount.fuse3"
+// helper binary is called.
 func (m *Mount) Mount(target string) error {
+	for _, helperBinary := range allowedHelperBinaries {
+		// helperBinary = "mount.fuse", typePrefix = "fuse."
+		typePrefix := strings.TrimPrefix(helperBinary, "mount.") + "."
+		if strings.HasPrefix(m.Type, typePrefix) {
+			return m.mountWithHelper(helperBinary, typePrefix, target)
+		}
+	}
```

Comment:
when FUSE is used, you should make sure Unmount uses `fusermount|fusermount3 -u`.

I've tried:

```
diff --git a/mount/mount_linux.go b/mount/mount_linux.go
index 83a086d0..a961031e 100644
--- a/mount/mount_linux.go
+++ b/mount/mount_linux.go
@@ -107,6 +107,7 @@ func Unmount(target string, flags int) error {
 }
 
 func unmount(target string, flags int) error {
+       exec.Command("fusermount3", "-u", target).Run()
```

and it solves the issue reported here: https://github.com/containers/fuse-overlayfs/issues/134

Reply:
thanks, applied